src/frontend/profile_utils.rs

```diff
@@ -0,0 +1,34 @@
+use std::time::{Duration, Instant};
+
+// Simple timer intended for profiling
+// Lazily inits. Returns `Duration`s or raw string (in ms)
+// Use Option as a safeguard because apparently previous Rust versions panicked
+// when current time was earlier than self.
```

Comment:
Minor:

[rustdoc comments should use triple-slashes](https://doc.rust-lang.org/rustdoc/how-to-write-documentation.html)

```suggestion
/// Simple timer intended for profiling
/// Lazily inits. Returns `Duration`s or raw string (in ms)
/// Use Option as a safeguard because apparently previous Rust versions panicked
/// when current time was earlier than self.
```

Reply:
Done, thanks!